src/csync/vio/csync_vio_local_unix.cpp

```diff
@@ -83,31 +84,26 @@ int csync_vio_local_closedir(csync_vio_handle_t *dhandle) {
   return rc;
 }
 
-csync_vio_file_stat_t *csync_vio_local_readdir(csync_vio_handle_t *dhandle) {
+
+std::unique_ptr<csync_file_stat_t> csync_vio_local_readdir(csync_vio_handle_t *dhandle) {
 
   dhandle_t *handle = NULL;
-  csync_vio_file_stat_t *file_stat = NULL;
 
   handle = (dhandle_t *) dhandle;
   struct _tdirent *dirent = NULL;
-
-  errno = 0;
```

Comment:
This must stay i think...   or?

Reply:
Looks like this came from https://github.com/owncloud/client/commit/c281a720b030fc52c22036f10fa2d2f3180e4359 and can probably go since we don't rely on errno to detect errors anymore.